lib/Server.js

```diff
@@ -130,12 +130,13 @@ class Server {
         next(err);
       });
 
-      return new Promise(resolve => {
+      const listenPromise = new Promise(resolve => {
         this.httpServer = this.server.listen(this.port, () => {
           console.log(`Server listening at http://${this.hostname}:${this.port}`);
           resolve();
         });
       });
+      return Promise.all([listenPromise, readyPromise]);
```

Comment:
My understanding is the Promise will return immediately. And unless the
async/await pattern, or generator functions are used, the net result will
be the same WRT timing of the two servers starting. I could be wrong, but
it seems the only functional change is the timeout, which I'd argue is the
responsibility of the CI system.
On Fri, Nov 27, 2015 at 9:01 PM Philipp von Weitershausen <
notifications@github.com> wrote:

> In lib/Server.js
> <https://github.com/mjohnston/react-native-webpack-server/pull/144#discussion_r46075231>
> :
>
> >          this.httpServer = this.server.listen(this.port, () => {
> >            console.log(`Server listening at http://${this.hostname}:${this.port}`);
> >            resolve();
> >          });
> >        });
> > +      return Promise.all([listenPromise, readyPromise]);
>
> The reason for having Server.start() not resolve its promise until all
> servers are up and running is so that the HTTP request made by
> createBundle() doesn't time out while the RN packager and webpack are
> still traversing the file system for their dependencies. On slow machines
> this can take quite a while. With this change, the HTTP request doesn't get
> made until both packagers are ready to receive requests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mjohnston/react-native-webpack-server/pull/144/files#r46075231>
> .
>